pyat/at/lattice/lattice_info.py

```diff
@@ -0,0 +1,100 @@
+from warnings import warn
+from math import pi
+import numpy
+from scipy.constants import physical_constants as cst
+import at       # for AtWarning, AtError
+from ..lattice import checktype, elements
+
+__all__ = ['get_energy', 'get_voltage', 'get_energy_loss']
+
+TWO_PI_ERROR = 1.E-4
+
+
+def get_energy(ring):
+    """
+    energy, periodicity = get_energy(ring) Get the ring energy
+    get_energy looks for the machine energy in:
+        1) the 1st 'RingParam' element
```

Comment:
I don't particularly like the RingParam element idea. It's not part of the lattice and would belong better in a lattice object. I'm happy to keep it for backwards compatibility with existing lattices though.

Reply:
I agree with the idea of a lattice object, however in MATLAB the RingParam element is still necessary (before it was introduced, AT used global variables, even worse). So let’s keep it for compatibility.

Reply:
Excuse me if I'm missing something but, Is it not worth checking at least some of the other elements, rather than raising an error after only checking for RingParams and RFCavities? As an example, our dba ring doesn't have either but it does have Energy attached to all non-drift elements.

Reply:
@T-Nicholls 
short answer: I just copied the Matlab behaviour, but I can look for energy wherever it can be!

full answer: Energy is not an element attribute, so the good practice should be to set it in a single place, Cavity element, possibly inactive, or even better RingParam element. However the `Energy` attribute is necessary for all '*RadPass' passmethods. That's why`at.radiation_on` silently adds it to all radiating elements. Coming back to your example, a ring with radiation but no cavity would not work. And if it is not radiating (no *RadPass pass method in dipoles), the energy attribute is useless, though it is perfectly valid. So following @willrogers's idea, here is what I propose:

I will create a lattice object, such as all existing functions can be kept unchanged, working with any sequence of elements, including the new lattice object. The `get_energy` function disappears, replaced by the `energy` attribute of the lattice object. So functions requiring the energy (like `ohmi_envelope`)will need a lattice object as input rather than a simple sequence. At instantiation, the lattice object will look for energy wherever it is, RingParam, Cavity or anything else.

Reply:
@T-Nicholls 
short answer: I just copied the Matlab behaviour, but I can look for energy wherever it can be!

full answer: Energy is not an element attribute, so the good practice should be to set it in a single place, Cavity element, possibly inactive, or even better RingParam element. However the `Energy` attribute is necessary for all '*RadPass' passmethods. That's why`at.radiation_on` silently adds it to all radiating elements. Coming back to your example, a ring with radiation but no cavity would not work. And if it is not radiating (no *RadPass pass method in dipoles), the energy attribute is useless, though it is perfectly valid. So following @willrogers's idea, here is what I propose:

I will create a lattice object, such as all existing functions can be kept unchanged, working with any sequence of elements, including the new lattice object. The `get_energy` function disappears, replaced by the `energy` attribute of the lattice object. So functions requiring the energy (like `ohmi_envelope`)will need a lattice object as input rather than a simple sequence. At instantiation, the lattice object will look for energy wherever it is, RingParam, Cavity or anything else.

Reply:
@lfarv, thank you for the clear explanation. I think that a lattice object is a good idea, and will be beneficial in other areas too.